src/components/visualizations/chart/legend/helpers.ts

```diff
@@ -224,37 +282,50 @@ export function buildHeatmapLabelsConfig(labels: string[], config: any) {
+function getColorLegendLabelsConfiguration(legendLabels: string[], isSmall: boolean, isVertical: boolean) {
+    const numberOfLabels = legendLabels.length;
     const firstLabelLength = head(legendLabels).length;
     const lastLabelLength = last(legendLabels).length;
     const maxLabelLength = firstLabelLength > lastLabelLength ? firstLabelLength : lastLabelLength;
     const labelLengths = isSmall ? SMALL_LABEL_LENGHT_THRESHOLDS : LABEL_LENGHT_THRESHOLDS;
 
-    let shorteningLevel: number;
-    let shorteningConfig;
+    const shorteningConfig = isVertical
+        ? verticalHeatmapConfig
+        : getHorizonalShorteningLabelConfig(labelLengths, maxLabelLength, isSmall, numberOfLabels);
 
-    if (isVertical) {
-        shorteningConfig = verticalHeatmapConfig;
-    } else {
-        if (inRange(maxLabelLength, 0, labelLengths[0])) {
-            shorteningLevel = 5;
-        } else if (inRange(maxLabelLength, labelLengths[0], labelLengths[1])) {
-            shorteningLevel = 4;
-        } else if (inRange(maxLabelLength, labelLengths[1], labelLengths[2])) {
-            shorteningLevel = 3;
-        } else if (inRange(maxLabelLength, labelLengths[2], labelLengths[3])) {
-            shorteningLevel = 2;
-        } else if (inRange(maxLabelLength, labelLengths[3], labelLengths[4])) {
-            shorteningLevel = 1;
-        } else if (maxLabelLength > labelLengths[4]) {
-            shorteningLevel = 0;
-        }
-        shorteningConfig = isSmall
-            ? heatmapSmallLegendConfigMatrix[shorteningLevel]
-            : heatmapLegendConfigMatrix[shorteningLevel];
-    }
+    return buildColorLabelsConfig(legendLabels, shorteningConfig);
+}
 
-    return buildHeatmapLabelsConfig(legendLabels, shorteningConfig);
+function getHorizonalShorteningLabelConfig(
+    labelLengths: number[],
+    maxLabelLength: number,
+    isSmall: boolean,
+    numberOfLabels: number,
+) {
+    const shorteningLevel = getColorLabelShorteningLevel(labelLengths, maxLabelLength);
+    return isSmall
+        ? heatmapSmallLegendConfigMatrix[shorteningLevel]
+        : numberOfLabels === 8
+        ? heatmapLegendConfigMatrix[shorteningLevel]
+        : colorLegendConfigMatrix[shorteningLevel];
+}
+
+function getColorLabelShorteningLevel(labelLengths: number[], maxLabelLength: number): number {
+    let shorteningLevel: number;
+    if (inRange(maxLabelLength, 0, labelLengths[0])) {
+        shorteningLevel = 5;
+    } else if (inRange(maxLabelLength, labelLengths[0], labelLengths[1])) {
+        shorteningLevel = 4;
+    } else if (inRange(maxLabelLength, labelLengths[1], labelLengths[2])) {
+        shorteningLevel = 3;
+    } else if (inRange(maxLabelLength, labelLengths[2], labelLengths[3])) {
+        shorteningLevel = 2;
+    } else if (inRange(maxLabelLength, labelLengths[3], labelLengths[4])) {
+        shorteningLevel = 1;
+    } else if (maxLabelLength > labelLengths[4]) {
+        shorteningLevel = 0;
+    }
+    return shorteningLevel;
```

Comment:
So do we intend to return default `undefined` if no `if`s is satisfied?

Reply:
done, change from `else if`  to `else`